api/v1beta1/foundationdbcluster_types.go

```diff
@@ -2390,6 +2403,27 @@ type ServiceConfig struct {
 	PublicIPSource *PublicIPSource `json:"publicIPSource,omitempty"`
 }
 
+// RoutingConfig allows configuring routing to our pods, and services that sit
+// in front of them.
+type RoutingConfig struct {
+	// Headless determines whether we want to run a headless service for the
+	// cluster.
+	HeadlessService *bool `json:"headlessService,omitempty"`
```

Comment:
Strictly speaking we never use the headless service for routing.

Reply:
Yeah, but I felt that "Routing" was closer than "Service" for this group of settings, since in the long term the plan is to use the headless service for discovery. Do you have any other suggestions for what we could call this?

Reply:
I felt like `Routing` captured the intent of these fields better than `Service`, but I'm open to other suggestions about how to make the meaning clearer.

Reply:
What about `Discovery`? I mean that's also an overloaded term. Otherwise I don't have any better idea :(  Like always:

"There are 2 hard problems in computer science: cache invalidation, naming things, and off-by-1 errors." :)